workflow-service/src/test/java/com/redhat/parodos/workflow/registry/BeanWorkFlowRegistryImplTest.java

```diff
@@ -0,0 +1,166 @@
+package com.redhat.parodos.workflow.registry;
+
+import com.redhat.parodos.workflow.WorkFlowType;
+import com.redhat.parodos.workflow.definition.service.WorkFlowDefinitionServiceImpl;
+import com.redhat.parodos.workflow.task.WorkFlowTask;
+import com.redhat.parodos.workflows.work.Work;
+import com.redhat.parodos.workflows.workflow.SequentialFlow;
+import com.redhat.parodos.workflows.workflow.WorkFlow;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BeanWorkFlowRegistryImplTest {
+
+	private ConfigurableListableBeanFactory beanFactory;
+
+	private WorkFlowDefinitionServiceImpl workFlowDefinitionService;
+
+	@BeforeEach
+	public void initEach() {
+		this.beanFactory = Mockito.mock(ConfigurableListableBeanFactory.class);
+		this.workFlowDefinitionService = Mockito.mock(WorkFlowDefinitionServiceImpl.class);
+	}
+
+	@Test
+	void simpleTestWithoutWorkflows() {
+		// when
+		BeanWorkFlowRegistryImpl registry = new BeanWorkFlowRegistryImpl(this.beanFactory,
+				this.workFlowDefinitionService, new HashMap<String, WorkFlowTask>(), new HashMap<String, WorkFlow>());
```

Comment:
I'd like to suggest to use diamonds instead of explicit arguments

```
this.workFlowDefinitionService, new HashMap<>(), new HashMap<>());
```

This also applies to other `new BeanWorkFlowRegistryImpl(....)`